## NetFilx Analysis

In this analysis we are gonna know about the netflix data set.

(IMPACT)

Which will help to get more insights form the data compare to the simple tabular data or csv file.

So,Lets Start

# 1) Getting the DataSet

There are many ways of getting the data. It can API Call , Online Posted , Web Scraping

For Netfilx analysis we are gonna use Online posted Dataset from the KAGGLE

Importing the library for getting data and wrangling (Pandas) also importing (matplotlib),(seaborn) and (ploty) for data visualisation

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

We are gonna use the data

In [2]:
data=pd.read_csv("netflix_titles[1].csv")

Coverting data to dataframe

In [3]:
df=pd.DataFrame(data)

DataFrame Preview

In [ ]:
df.sample(5)

show_id     type                     title            director  \
6761   s6762    Movie                Fishpeople        Keith Malloy   
4018   s4019    Movie  Barcelona: A Love Untold   Olivia M. Lamasan   
3927   s3928  TV Show                  New Girl                 NaN   
8699   s8700    Movie                Warehoused  Jack Zagha Kababie   
6813   s6814    Movie              Frozen River       Courtney Hunt   

                                                   cast        country  \
6761                                                NaN            NaN   
4018  Kathryn Bernardo, Daniel Padilla, Joshua Garci...    Philippines   
3927  Zooey Deschanel, Jake Johnson, Max Greenfield,...  United States   
8699                    José Carlos Ruiz, Hoze Meléndez         Mexico   
6813  Melissa Leo, Misty Upham, Charlie McDermott, M...  United States   

            date_added  release_year rating   duration  \
6761     April 1, 2018          2017  TV-14     49 min   
4018    March 14, 2019          2016  TV-14    132 min   
3927    April 11, 2019          2018  TV-14  7 Seasons   
8699  November 1, 2017          2015  TV-14     92 min   
6813      July 1, 2019          2008      R     97 min   

                                          listed_in  \
6761                   Documentaries, Sports Movies   
4018  Dramas, International Movies, Romantic Movies   
3927                 Romantic TV Shows, TV Comedies   
8699         Comedies, Dramas, International Movies   
6813                     Dramas, Independent Movies   

                                            description  
6761  In this exploration of the life-changing power...  
4018  While pursuing a degree in Spain, an architect...  
3927  Still rebounding from a breakup, Jessica Day m...  
8699  A soon-to-be-retiring Mr. Lino teaches 20-some...  
6813  On a Mohawk reservation on the Canadian border...

## Info about the data

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   show_id       8807 non-null   str  
 1   type          8807 non-null   str  
 2   title         8807 non-null   str  
 3   director      6173 non-null   str  
 4   cast          7982 non-null   str  
 5   country       7976 non-null   str  
 6   date_added    8797 non-null   str  
 7   release_year  8807 non-null   int64
 8   rating        8803 non-null   str  
 9   duration      8804 non-null   str  
 10  listed_in     8807 non-null   str  
 11  description   8807 non-null   str  
dtypes: int64(1), str(11)
memory usage: 825.8 KB


## Size and Shape of dataset

* Shape - Number of rows and columns.
* Size -Total number of value present in dataset

In [13]:
size=df.size
print("The total values in dataset is : ",size)

The total values in dataset is :  105684


In [14]:
shape=df.shape
print("The value of rows and coluns is : ",shape)

The value of rows and coluns is :  (8807, 12)


## Checking the null values

In [16]:
NUll=df.isnull().sum()
NUll

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

WHy do we check null values?

Because by checking null value we are able to sort the data means we can drop the larger number of null values column from the dataset.
eg. director column have 2634 null values which are higher not that much because we have 8807 rows , so we can drop it if we want


## Handling the null values

* Director , country and cast can be replace by - Unknown 
* date can be replace by - median value
* rating and duration  can be replace by - Not rated and none


In [9]:
df.fillna({"director": "Unknown","cast":"Unknown","country":"Unknown"}, inplace=True)
df.director.isnull().sum(),df.cast.isnull().sum(),df.country.isnull().sum()

(np.int64(0), np.int64(0), np.int64(0))

In [89]:
df["date_added"]=pd.to_datetime(df["date_added"],format="%B %d, %Y")
df.fillna({"date_added":df["date_added"].median()},inplace=True)
df.duration.isnull().sum()

np.int64(0)

In [78]:
df.fillna({"rating":"Not rated"},inplace=True)
df.loc[df["rating"]=="Not rated"]
df["rating"].isnull().sum()

np.int64(0)

In [84]:
df.fillna({"duration":"none"},inplace=True)
df.duration.isnull().sum()

np.int64(0)

Checking is still there null values

In [90]:
df.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

NO NULL VALUES

# 2) Start with EDA

## Univarient Analysis

### Content Type Distribution

Movie vs TV Show count

In [26]:
Movie_count=(df["type"]=="Movie").sum()
Show_count=(df["type"]!="Movie").sum()
print("The number of movie is : ",Movie_count)
print("The number of show is : ",Show_count)

The number of movie is :  6131
The number of show is :  2676


Visualise

In [25]:
count=pd.DataFrame(df["type"].value_counts())
fig=px.bar(count,title="Movies VS Shows")
fig.update_layout(font=dict(size=15,color="grey"),title_font=dict(size=20,color="black"))
fig



### Rating Distribution

In [68]:
rating_count=pd.DataFrame(df["rating"].value_counts().reset_index())
rating_count

rating  count
0      TV-MA   3207
1      TV-14   2160
2      TV-PG    863
3          R    799
4      PG-13    490
5      TV-Y7    334
6       TV-Y    307
7         PG    287
8       TV-G    220
9         NR     83
10         G     41
11  TV-Y7-FV      6
12     NC-17      3
13        UR      3

In [ ]:
fig=px.bar(rating_count,x="rating",y="count",title="Distribution of Ratings")
fig.update_layout(title_font=dict(size=20,color="black"),font=dict(size=14 , color="grey"))
fig


### Release Year Distribution

In [61]:
rel_count=pd.DataFrame(df["release_year"].value_counts().reset_index())
rel_count

release_year  count
0           2018   1147
1           2017   1032
2           2019   1030
3           2020    953
4           2016    902
..           ...    ...
69          1961      1
70          1959      1
71          1925      1
72          1966      1
73          1947      1

[74 rows x 2 columns]

In [ ]:
fig=px.histogram(rel_count,x="release_year",y="count")
fig

## Country Analysis

In [77]:
top10=df.explode("country")["country"].value_counts().reset_index().head(10)
top10

country  count
0   United States   2818
1           India    972
2  United Kingdom    419
3           Japan    245
4     South Korea    199
5          Canada    181
6           Spain    145
7          France    124
8          Mexico    110
9           Egypt    106

In [80]:
fig=px.bar(top10,x="country",y="count",title="Top 10 countries in netflix")
fig.update_layout(title_font=dict(size=20,color="black"))
fig

## Genre Analysis

### Top genres

In [98]:
Top5=df["listed_in"].str.split(",").explode().value_counts().reset_index().head(5)
Top5

listed_in  count
0   International Movies   2624
1                 Dramas   1600
2               Comedies   1210
3     Action & Adventure    859
4          Documentaries    829

In [99]:
fig=px.bar(Top5,x="listed_in",y="count")
fig

### Genre vs Content Type

#### Top genre for movies

In [8]:
on_movie=df[df["type"]=="Movie"]
list_movie=on_movie["listed_in"].str.split(",").explode().str.strip().value_counts().reset_index()
top10_movie_gen=list_movie.head(10)
top10_movie_gen

listed_in  count
0      International Movies   2752
1                    Dramas   2427
2                  Comedies   1674
3             Documentaries    869
4        Action & Adventure    859
5        Independent Movies    756
6  Children & Family Movies    641
7           Romantic Movies    616
8                 Thrillers    577
9          Music & Musicals    375

##### visualise

In [54]:
fig=px.bar(top10,x="listed_in",y="count",title="Top 10 genres in movies")
fig.update_layout(dict(title_font=dict(color="black",size=20)))
fig

#### Top genre for Shows

In [9]:
onli_show=df[df["type"]=="TV Show"]
show_gen=onli_show["listed_in"].str.split(",").explode().str.strip().value_counts().reset_index()
top10_show_gen=show_gen.head(10)
top10_show_gen

listed_in  count
0  International TV Shows   1351
1               TV Dramas    763
2             TV Comedies    581
3          Crime TV Shows    470
4                Kids' TV    451
5              Docuseries    395
6       Romantic TV Shows    370
7              Reality TV    255
8        British TV Shows    253
9            Anime Series    176

##### visualise

In [51]:
fig=px.bar(top10_show_gen,x="listed_in",y="count",title="Top 10 genres in Tv show")
fig.update_layout(title_font=dict(color="Black",size=20))
fig

## Date Added Analysis

As we know that in dataframe the date column have string datatype so we have to convert it into date_time datatype

In [5]:
df["date_added"]=pd.to_datetime(df["date_added"],errors="coerce")

Extracting the year from date

In [6]:
df["year"]=df["date_added"].dt.year

### Which year highest content added

#### For Movies

In [10]:
#Only Movies
on_movie["date_added"]=pd.to_datetime(on_movie["date_added"],errors="coerce")
on_movie["year"]=on_movie["date_added"].dt.year
top10_yr_mov=on_movie.groupby("year")["type"].value_counts().reset_index()[["year","count"]].sort_values(by="year",ascending=False).head(10)
top10_yr_mov

year  count
13  2021    993
12  2020   1284
11  2019   1424
10  2018   1237
9   2017    839
8   2016    253
7   2015     56
6   2014     19
5   2013      6
4   2012      3

In [11]:
fig=px.bar(top10_yr_mov,x="year",y="count",title="Top 10 years having Highest content added")
fig

#### For Tvshows

In [12]:
onli_show["date_added"]=pd.to_datetime(onli_show["date_added"],errors="coerce")
onli_show["year"]=onli_show["date_added"].dt.year
top10_yr_show=onli_show.groupby("year")["type"].value_counts().reset_index()[["year","count"]].sort_values(by="count",ascending=False)
top10_yr_show

year  count
8  2020.0    594
7  2019.0    575
9  2021.0    505
6  2018.0    388
5  2017.0    325
4  2016.0    165
3  2015.0     17
1  2013.0      4
2  2014.0      4
0  2008.0      1

In [11]:
fig=px.bar(top10_yr_show,x="year",y="count")
fig

#### Month wise analysis

Enter the type and year for which you want the data. This will display the month-wise number of content for that particular year and type.        

In [47]:
sel_year=int(input("Enter year : " ))
sel_type=input("Enter the type (Movie or TV Show): ")

df_sel=df.loc[(df["date_added"].dt.year==sel_year)&(df["type"]==sel_type)]
df_sel["date_added"]=pd.to_datetime(df_sel["date_added"])

#getting the month
df_sel["month"]=df_sel["date_added"].dt.month_name()
sel_data_of_year=df_sel.groupby("month")["type"].value_counts().reset_index()[["month","count"]]
sel_data_of_year

month  count
0     April      4
1  December      4
2      July      2
3      June      1
4     March      1
5  November      1
6   October      4

In [48]:
fig=px.bar(sel_data_of_year,x="month",y="count",title=f"Content added in {sel_year} of type {sel_type} according the months")
fig

## Director & Cast Analysis

#### Top 10 directors

In [28]:
# Filter data of Known Directors only
dir_type=input("Enter the type : ")

Dir_data=df.loc[(df["director"]!="Unknown")&(df["type"]==dir_type)]
Dir_data=Dir_data["director"].value_counts().reset_index().head(10)
Dir_data

director  count
0  Alastair Fothergill      3
1          Hsu Fu-chun      2
2      Rob Seidenglanz      2
3          Shin Won-ho      2
4          Stan Lathan      2
5       Iginio Straffi      2
6            Ken Burns      2
7      Julien Leclercq      1
8        Mike Flanagan      1
9      Andy Devonshire      1

Visualisation

In [29]:
fig=px.bar(Dir_data,x="director",y="count",title=f"Top10 Director in a {dir_type}")
fig

#### Most frequent actors

In [40]:
# type_=input("Enter the type : ")
cast_data=df["cast"].str.split(",").explode().reset_index()
# filter data of cast without unknown
# cast_data=cast_data.loc[(cast_data["cast"]="Unknown")&(cast_data["type"])]
# cast_data["cast"].value_counts().reset_index().head(10)
cast_data

index                    cast
0          0                 Unknown
1          1              Ama Qamata
2          1             Khosi Ngema
3          1           Gail Mabalane
4          1          Thabang Molaba
...      ...                     ...
64946   8806        Manish Chaudhary
64947   8806            Meghna Malik
64948   8806           Malkeet Rauni
64949   8806          Anita Shabdish
64950   8806   Chittaranjan Tripathy

[64951 rows x 2 columns]